File: ios/LookSeeProto/Models/NearbyLandmark.swift
//
//  NearbyLandmark.swift
//  LookSeeProto
//
//  Created by Ian Thompson on 3/9/26.
//

import Foundation

struct NearbyLandmark: Codable, Identifiable, Hashable {
    let landmarkId: String
    let label: String
    let shortDescription: String
    let latitude: Double
    let longitude: Double
    let distanceMeters: Double
    
    // MARK: - New Filter Fields (Matched to the updated Lambda)
    let createdBy: String?
    let createdAt: String?
    let promotionEnabled: Bool
    let promotion: String?
    let clusterId: String? // <-- THIS IS THE NEWEST ADDITION!

    // 🚀 NEW: Rich UI fields pulled from the backend
    let websiteUrl: String?
    let promoName: String?        
    let promoDescription: String?
    let promoImageUrl: String?
    let merchantName: String?
    let merchantBio: String?
    let merchantPhone: String?
    let merchantAddress: String?
    let merchantLogoUrl: String?

    var id: String { landmarkId }
}

struct NearbyLandmarksRequest: Codable {
    let latitude: Double
    let longitude: Double
    let radiusMeters: Double
    let limit: Int? // Added to support our new database limit safeguard!
}

struct NearbyLandmarksResponse: Codable {
    let items: [NearbyLandmark]
    let count: Int
    let radiusMeters: Double
}

File: Android/LookSee/app/src/main/java/looksee/angelll/com/models/NearbyLandmark.kt
package looksee.angelll.com.models

data class NearbyLandmark(
    val landmarkId: String = "",
    val label: String = "",
    val shortDescription: String = "",
    val latitude: Double = 0.0,
    val longitude: Double = 0.0,
    val distanceMeters: Double = 0.0,
    val createdBy: String? = null,
    val createdAt: String? = null,
    val promotionEnabled: Boolean = false,
    val promotion: String? = null,
    val clusterId: String? = null,
) {
    val id: String
        get() = landmarkId
}

data class NearbyLandmarksRequest(
    val latitude: Double,
    val longitude: Double,
    val radiusMeters: Double,
    val limit: Int? = 100,
)

data class NearbyLandmarksResponse(
    val items: List<NearbyLandmark> = emptyList(),
    val count: Int = items.size,
    val radiusMeters: Double = 0.0,
)
